### ISSUE TITLE
error when using '--builder'

### ISSUE THREAD

**Issue by slightlyoffbeat** [kss-node/kss-node#325]
using v3 beta 14

I ran

`kss --clone builder
`
no problems.

Then I run:

`kss src/scss --builder builder
`
I get:

`kss expected the builder to implement KssBuilderBase API version 3.0; version "3.0" is being used instead.`

**Comment by slightlyoffbeat** [comment]
OK. On my project I had install kss locally (also had it installed globally).

Not sure if that is considered a bug or user error? Working for me now.

**Comment by JohnAlbin** [comment]
That is an odd error.

**Comment by JohnAlbin** [comment]
hmm… I think I see the issue. If the builder in your `builder/` directory was loading a different version of kss-node then the one that was being run on the command line, then the check to see if they are instanceOf the same class will fail.

**Comment by JohnAlbin** [comment]
I'm not sure I can fix it though.

Regardless, I would suggest, @slightlyoffbeat, that you not install Node.js CLI utilities globally in most cases, because you should want the ability to run different versions of those tools on different projects.

I've written a HOWTO about this in the wiki: https://github.com/kss-node/kss-node/wiki/npm-quick-start-guide (specifically look at the npm-run command.)

**Comment by zaccolley** [comment]
Have been getting this error when using `kss()` on a `gulp.watch` but not when running the task directly.

**Comment by zaccolley** [comment]
I think it's to do with the path, but I've been debugging and it seems like the path to the builder directory is fine both times. Will keep looking!

**Comment by slightlyoffbeat** [comment]
Thanks. I use nvm, so I'm set on running diff versions of node in diff projects. If it helps, I think the kss-node version I had installed locally and globally were the same.

**Comment by zaccolley** [comment]
OK so when I output the error found at https://github.com/kss-node/kss-node/blob/master/builder/base/kss_builder_base.js#L162. I got `caught { [Error: Cannot find module '/path/to/project/builder'] code: 'MODULE_NOT_FOUND' }
`.

So it turned out the first time it was being run it was ignoring the `builder.js` file cause it couldn't find it.

When doing the `kss --clone builder` command, it generates a `builder.js` file. When I renamed the `builder.js` file to `index.js, it then recognised that file and used it, meaning it now works.

Is the file supposed to be named `builder.js`? Maybe I missed in the documentation this file

**Comment by JohnAlbin** [comment]
> When doing the kss --clone builder command, it generates a builder.js file. When I renamed the builder.js file to `index.js, it then recognised that file and used it, meaning it now works.

Yes, but it should also create a package.json file which has a `main` property pointing at `builder.js`. Does your cloned builder not have that package.json file?

**Comment by pinkkis** [comment]
I've only been able to make kss work with a global install. the local installs give that KssBuilderBase API version error, and i'm not even cloning a builder. just out of the box npm install into an empty project, doesn't work (no global install exists during the test).

node 6.10 and i've tried kss 3 beta 10-18

**Comment by Skadi2k3** [comment]
I'm using a custom twig builder and had to set the first require('kss/builder/base/twig') in builder.js to the local copy of kss, as it would otherwise use the globally installed kss, resulting in two different KssBuilderBase Classes being used and breaking the instanceof comparison in kss_builder_base*.js
